graph/network/maximum_flow.go

```diff
@@ -0,0 +1,168 @@
+package network
+
+import (
+	"container/list"
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+type edgeKey struct{ from, to int64 }
+
+type residualGraph struct {
+	Graph *simple.WeightedDirectedGraph
+	Flow  map[edgeKey]float64
+}
+
+func initializeResidualGraph(originalGraph graph.WeightedDirected) (*residualGraph, error) {
+	graphCopy := simple.NewWeightedDirectedGraph(0, 0)
+	flow := make(map[edgeKey]float64)
+	// Add all nodes
+	for nodes := originalGraph.Nodes(); nodes.Next(); {
+		graphCopy.AddNode(nodes.Node())
+	}
```

Comment:
```suggestion
	r := &residualGraph{
		g:    simple.NewWeightedDirectedGraph(0, 0)},
		flow: make(map[edgeKey]float64),
	}
	nodes := originalGraph.Nodes()
	for nodes.Next() {
		r.g.AddNode(nodes.Node())
	}
```
similar changes below